provider_gsuite.go

```diff
@@ -0,0 +1,217 @@
+package jwtauth
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/mitchellh/mapstructure"
+	"golang.org/x/oauth2/google"
+	"golang.org/x/oauth2/jwt"
+	admin "google.golang.org/api/admin/directory/v1"
+	"google.golang.org/api/option"
+)
+
+// GSuiteProvider provides G Suite-specific configuration and behavior.
+type GSuiteProvider struct {
+	// Configuration for the provider
+	config GSuiteProviderConfig
+	// Google JWT configuration
+	jwtConfig *jwt.Config
+	// Google admin service
+	adminSvc *admin.Service
+	// Context for requests to Google admin APIs
+	ctx context.Context
```

Comment:
Is this needed? Contexts are rarely stored on `structs` (or anywhere besided the call stack, really).

Reply:
Nope, this isn't needed. I've removed it in 2dd71b8549acc71e0be8277eefb32e05aa824831. Passing it through function calls instead.